src/renderer/components/NodeDocumentation/NodesList.tsx

```diff
@@ -1,18 +1,43 @@
 import { Box, Center, HStack, Text, VStack } from '@chakra-ui/react';
+import MiniSearch from 'minisearch';
 import { memo, useEffect, useMemo, useRef, useState } from 'react';
 import { useContext } from 'use-context-selector';
 import { BackendContext } from '../../contexts/BackendContext';
 import { NodeDocumentationContext } from '../../contexts/NodeDocumentationContext';
 import { getNodesByCategory } from '../../helpers/nodeSearchFuncs';
 import { IconFactory } from '../CustomIcons';
 
-export const NodesList = memo(() => {
+export const NodesList = memo(({ searchQuery }: { searchQuery: string }) => {
     const { selectedSchemaId, isOpen, openNodeDocumentation } =
         useContext(NodeDocumentationContext);
 
     const { schemata, categories } = useContext(BackendContext);
     const schema = schemata.schemata;
-    const byCategories = useMemo(() => getNodesByCategory(schema), [schema]);
+
+    const miniSearch = new MiniSearch({
+        idField: 'schemaId',
+        fields: ['category', 'description', 'name', 'subcategory'],
+    });
+
+    miniSearch.addAll(schema);
+
+    const searchResult = miniSearch.search(searchQuery, {
+        boost: { name: 2 },
+        fuzzy: 0.2,
+        prefix: true,
+        combineWith: 'AND',
+    });
+    const matchingSchemaIds = useMemo(
+        () => [...new Set(searchResult.map((s) => String(s.id)))],
```

Comment:
How are duplicates possible? Does mini search not guarantee that ids are unique in the search result list?

Reply:
Honestly the set thing was a copilot suggestion and I didn't even question it...